dist/ramda.js

```diff
@@ -4440,6 +4440,43 @@
         }, [], fns);
     };
 
+    // The algorithm used to handle cyclic structures is heavily
+    // inspired by underscore.js' implementation of isEqual
+    var _eqDeep = function _eqDeep(a, b, stackA, stackB) {
+        var idx = stackA.length;
+        while (--idx >= 0) {
+            if (stackA[idx] === a) {
+                return stackB[idx] === b;
+            }
+        }
+        stackA.push(a);
+        stackB.push(b);
+        var result;
+        switch (type(a)) {
+        case 'Array':
```

Comment:
> probably should also handle `Dates` and typed arrays

Are you thinking we should do so with the same implementation this uses for Objects and Arrays?  Obviously that wouldn't work for `Date`s, which have no enumerable properties.  But something like

    date1 - date2 === 0

or

    date1.getTime() === date2.getTime()

would also not capture these differences, if they're important:

    var date1 = new Date(), date2 = new Date();
    date1.color = 'red'; date2.color = 'blue';

Do you have an implementation in mind?